test/Scientist.Test/FireAndForgetResultPublisherTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub;
+using NSubstitute;
+using UnitTests;
+using Xunit;
+
+[Collection("Tests Dependent Upon static ResultPublisher that can't run in parallel")]
+public class FireAndForgetResultPublisherTests
+{
+    [Fact]
+    public async Task PublishesAsynchronously()
+    {
+        const int expectedResult = 42;
+
+        // Create a new publisher that will delay all
+        // publishing to account for this test.
+        var innerPublisher = Substitute.For<IResultPublisher>();
+        innerPublisher.Publish(Arg.Any<Result<int, int>>())
+            .Returns(call => Task.Delay(100));
```

Comment:
I'm a little antsy about using delays for tests because they take a little bit of determinism away from the test. For example, some weird timing issue could in theory cause this test to fail.

Instead, I generally like to use something like a `ManualResetEvent` here to tightly control order of events. Then we could call `Signal` on the event before we `await whenPublished`. Thoughts?

One of the reasons I note this is this test runs significantly slower than all the others because of this artificial delay.

BTW, the same approach probably applies to `HandlesDelayedExceptionsThrownByInnerPublisher`